lib/puppet/provider/nssdb/certutil.rb

```diff
@@ -0,0 +1,43 @@
+Puppet::Type.type(:nssdb).provide(:certutil) do
+  commands :certutil => 'certutil'
+
+  def create
+    generate_nssdb
+  end
+
+  def destroy
+    destroy_nssdb
+  end
+
+  def exists?
+    certutil(
+      '-L',
+      '-d', resource[:nssdb_dir]
+    )
+  rescue Puppet::ExecutionFailure
+    Puppet.debug("Did not find existing NSS database at #{resource[:nssdb_dir]}")
+    return false
+  end
+
+  private
+
+  def generate_nssdb
+    certutil(
+      '-N',
+      '-d', resource[:nssdb_dir],
+      '-f', resource[:password_file]
+    )
+  rescue Puppet::ExecutionFailure => e
+    Puppet.err("Failed to generate new NSS database at #{resource[:nssdb_dir]} with password file #{resource[:password_file]}: #{e}")
+    return nil
+  end
+
+  def destroy_nssdb
+    files = resource[:nssdb_files]
+    files = [files] unless files.is_a?(Array)
+
+    files.each do |file|
+      File.rm(file)
+    end
+  end
```

Comment:
@ehelms you have requested an implementation of this method which does not require the user to pass the list of `nssdb_files` to the provider.

I'm not sure how you envision this, but did you have in mind performing an `rm -rf $nssdb_dir` ? What if the password file (as in the previous implementation) or some other files extra to the base NSS DB files are present in this directory (for example, backups)?

Reply:
If we are managing the nssdb directory (which I think we should since that is how it works) then anything *in* the directory should be considered managed by this resource and thus can be created, update and destroyed with the lifecycle of this resource. So if this type puts the password file in that directory then it should also remove it.